### PR TITLE
SAMZA-2763: Support worker JVM opts for Samza Beam portable mode

### PR DESCRIPTION
**Pull request by mynameborat**
**Summary**: Support JVM options for worker process in Samza Beam portable mode
Description: With portable mode support for Samza Beam, we want to tune and configure the JVM options for worker process. In this PR, we add support by introducing worker.opts configuration and autosizing integration support.

**Changes**:
- Added worker.opts configuration
- Add autosizing integration support for Xmx
- Updated configuration table and website

**API Changes**: None
**Usage Instructions**: worker.opts can be used similar to other samza application configuration although it only applies to Samza Beam portable execution mode and is ignored otherwise.
**Upgrade Instructions**: None

### CODE REVIEW

**Comment by amansinghoriginal** [comment]
LGTM

File: samza-core/src/main/java/org/apache/samza/config/ShellCommandConfig.java [hunk 1]
```diff
@@ -97,20 +100,16 @@ public String getCommand() {
...
-    }
-    return jvmOpts;
+    String taskOpts = get(ShellCommandConfig.TASK_JVM_OPTS);
+    String autosizingContainerMaxHeap = get(JobConfig.JOB_AUTOSIZING_CONTAINER_MAX_HEAP_MB);
+
+    return Optional.ofNullable(getFinalJvmOptions(taskOpts, autosizingContainerMaxHeap));
+  }
+
+  public Optional<String> getWorkerOpts() {
+    String autosizingWorkerHeapMb = get(JobConfig.JOB_AUTOSIZING_WORKER_MAX_HEAP_MB);
```
[the remark this answers is not in the document]
Comment (mynameborat) [reply samza-core/src/main/java/org/apache/samza/config/ShellCommandConfig.java:110]: We don't know what would be the ideal default. Leaving it to the JVM to infer this or use its own default for now.

File: samza-core/src/main/java/org/apache/samza/config/ShellCommandConfig.java [hunk 2]
```diff
@@ -97,20 +100,16 @@ public String getCommand() {
...
-      }
-    }
-    return jvmOpts;
+    String taskOpts = get(ShellCommandConfig.TASK_JVM_OPTS);
+    String autosizingContainerMaxHeap = get(JobConfig.JOB_AUTOSIZING_CONTAINER_MAX_HEAP_MB);
+
+    return Optional.ofNullable(getFinalJvmOptions(taskOpts, autosizingContainerMaxHeap));
+  }
+
+  public Optional<String> getWorkerOpts() {
```
[the remark this answers is not in the document]
Comment (mynameborat) [reply samza-core/src/main/java/org/apache/samza/config/ShellCommandConfig.java:109]: Done

**Review by shanthoosh — APPROVED** [review]
LGTM.

**Pull request merged**